docs/strategies/hedging.md

```diff
@@ -471,6 +471,45 @@ Some insights about the hedging callback behavior in relation to hedging context
 > [!NOTE]
 > The hedging strategy ensures that both `ActionGenerator` and `OnHedging` are never executed concurrently. Any modifications to the contexts are thread-safe in these callbacks.
 
+### Sequence diagram about contexts and callbacks
+
+For the sake of conciseness the `Caller` is not depicted on the diagram.
+
+```mermaid
+sequenceDiagram
+    autonumber
+    participant P as Pipeline
+    participant H as Hedging
+    participant AG as ActionGenerator
+    participant UC as UserCallback
+    participant HUC as HedgedUserCallback
+    participant OH as OnHedging
+
+    P->>H: Calls ExecuteCore <br/>with Primary Context
+    H-->>H: Deep clones <br/>Primary Context
+
+    H->>+UC: Invokes <br/>with Primary Context
+    UC-->>UC: Processes <br/>+ Modifies Context
+    UC->>-H: Fails
+
+    H-->>H: Deep clones <br/>Primary Context
+    H->>+AG: Invokes <br/>with both Contexts
+    AG-->>AG: Modifies Primary Context
```

Comment:
```suggestion
    AG-->>AG: Modifies Primary or Action Context 
```

you can also modify action context here.

Reply:
I've updated the `OnHedging` part also.